x-pack/plugin/slm/src/main/java/org/elasticsearch/xpack/slm/SLMGetExpiredSnapshotsAction.java

```diff
@@ -134,13 +134,13 @@ protected void doExecute(Task task, Request request, ActionListener<Response> li
                                 )
 
                                 // Compute snapshots to delete for each (relevant) policy
-                                .<Void>andThen((l, snapshotDetailsByPolicy) -> ActionListener.completeWith(l, () -> {
+                                .<Void>andThenCompleteWith(snapshotDetailsByPolicy -> {
```

Comment:
I'd like to specialize this case too by adding `andThenVoid`. This avoids both the `<Void>` case and the return null.

Reply:
I'd rather keep `andThenVoid` for an async version of this, which is much more common. I've tried `andThenConsume`, how about that?

Reply:
Sure, though if we use `andThenApply` above, it could also be `andThenAccept` to piggy-back on the method names of Function/Consumer. I am good either way.